packages/grafana-llm-frontend/src/mcp.tsx

```diff
@@ -259,25 +294,25 @@ export function MCPClientProvider({
   appVersion,
   children,
 }: MCPClientProviderProps) {
-  const resource = createClientResource(appName, appVersion);
+  const { value: resource } = useAsync(() => createClientResource(appName, appVersion));
 
   // This will either return the client or throw a promise/error.
   // If it throws a promise, Suspense will suspend the component until it resolves.
   // If it throws an error, it should be caught by an ErrorBoundary.
-  const client = resource.read();
+  const result = resource?.read();
 
   // Cleanup when the component unmounts.
   React.useEffect(() => {
     return () => {
-      if (client) {
-        client.close();
+      if (result?.client) {
+        result.client.close();
       }
       clientMap.delete(clientKey(appName, appVersion));
     };
-  }, [client, appName, appVersion]);
+  }, [result, appName, appVersion]);
 
   return (
-    <MCPClientContext.Provider value={client}>
+    <MCPClientContext.Provider value={result ?? null}>
```

Comment:
if we return `null` here would this not trigger the `if (client === null)` check on line 330?